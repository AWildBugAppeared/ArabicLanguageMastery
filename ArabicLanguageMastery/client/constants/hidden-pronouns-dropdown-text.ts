export const HiddenPronounsDropdownText = {
  faailPresent: 'No Hidden Faail',
  huwa: 'هو',
  hiya: 'هي',
  anta: 'أنت',
  ana: 'أنا',
  nahnu: 'نحن',
};
