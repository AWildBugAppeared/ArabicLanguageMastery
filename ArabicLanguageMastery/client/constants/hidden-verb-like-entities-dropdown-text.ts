export const HiddenVerbLikeEntitiesDropdownText = {
  shibhulFilPresent: 'No Hidden VLE',
  mawjuwdun: 'مَوْجُوْدٌ',
  mawjuwdan: 'مَوْجُوْدًا',
  mawjuwdin: 'مَوْجُوْدٍ',
  mawjuwdatun: 'مَوْجُوْدَةٌ',
  mawjuwdatan: 'مَوْجُوْدَةً',
  mawjuwdatin: 'مَوْجُوْدَةٍ',
};
