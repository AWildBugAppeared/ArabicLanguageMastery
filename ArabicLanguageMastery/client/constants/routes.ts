const introductoryModuleRoute = 'introductory-module';

export const Routes = {
  introductoryModule: {
    theArabicSciences: `/${introductoryModuleRoute}/the-arabic-sciences`,
    introductionToSarf: `/${introductoryModuleRoute}/introduction-to-sarf`,
    introductionToNahwPart1: `/${introductoryModuleRoute}/introduction-to-nahw-part-1-the-two-types-of-sentences`,
    theMaadhiVerb: `/${introductoryModuleRoute}/the-maadhi-verb`,
    introductionToPhrases: `/${introductoryModuleRoute}/introduction-to-phrases`,
    theDescriptivePhrase: `/${introductoryModuleRoute}/the-descriptive-phrase`,
    thePossessivePhrase: `/${introductoryModuleRoute}/the-possessive-phrase`,
    theThreePartsOfSpeech: `/${introductoryModuleRoute}/the-three-parts-of-speech`,
  },
  contactUs: '/contact-us',
  englishToArabic: '/english-to-arabic',
};
