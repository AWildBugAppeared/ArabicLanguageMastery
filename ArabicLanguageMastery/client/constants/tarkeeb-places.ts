export const TarkeebPlaces = {
  ismInna: 'اسم إن',
  ismKaana: 'اسم كان',
  jaar: 'جارّ',
  haal: 'حال',
  khabr: 'خبر',
  khabrInna: 'خبر إن',
  khabrKaana: 'خبر كان',
  dhulHaal: 'ذو الحال',
  atf: 'عطف',
  faail: 'فاعل',
  fil: 'فعل',
  mafoolBiHi: 'مفعول به',
  mafoolFiyHi: 'مفعول فيه',
  mafoolLaHu: 'مفعول له',
  mafoolMutlaq: 'مفعول مطلق',
  majroor: 'مجرور',
  matoof: 'معطوف',
  matoofAlayh: 'معطوف عليه',
  mubtada: 'مبتدأ',
  mudaaf: 'مضاف',
  mudaafIlayhi: 'مضاف إليه',
  mutaalliq: 'متعلّق',
};
