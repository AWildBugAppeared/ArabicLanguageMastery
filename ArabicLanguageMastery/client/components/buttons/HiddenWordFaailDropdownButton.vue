<template>
  <dropdown-button-base
    :button-classes="buttonClasses"
    :button-text="tarkeebPlace"
    :dropdown-text-list="faailDropdownTextList"
    @dropdownSelected="setFaail"
  ></dropdown-button-base>
</template>

<script lang="ts">
import Vue from 'vue';

import DropdownButtonBase from '~/components/buttons/DropdownButtonBase.vue';
import { HiddenPronounsDropdownText } from '~/constants/hidden-pronouns-dropdown-text';
import { TarkeebPlaces } from '~/constants/tarkeeb-places';

export default Vue.extend({
  name: 'HiddenWordFaailDropdownButton',

  components: { DropdownButtonBase },

  props: {
    buttonClasses: {
      type: String,
      required: false,
      default: '',
    },
  },

  data() {
    return {
      tarkeebPlace: TarkeebPlaces.fil,
      faailDropdownTextList: Object.values(HiddenPronounsDropdownText),
    };
  },

  methods: {
    setFaail(faail: string | null) {
      if (faail === HiddenPronounsDropdownText.faailPresent) {
        faail = null;
      }

      this.$emit('faailSet', faail);
    },
  },
});
</script>

<style lang="scss" scoped></style>
