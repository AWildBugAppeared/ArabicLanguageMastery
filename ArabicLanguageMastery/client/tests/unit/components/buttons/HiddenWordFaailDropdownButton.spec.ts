/* eslint-disable  @typescript-eslint/no-explicit-any */

// Imports
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';

import HiddenWordFaailDropdownButton from '@/components/buttons/HiddenWordFaailDropdownButton.vue';
import { HiddenPronounsDropdownText } from '~/constants/hidden-pronouns-dropdown-text';
import { TarkeebPlaces } from '~/constants/tarkeeb-places';

describe('HiddenWordFaailDropdownButton.vue', () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let vuetify: Vuetify;

  const mountWrapper = () => {
    wrapper = mount(HiddenWordFaailDropdownButton, {
      localVue,
      vuetify,
      propsData: {
        buttonClasses: 'primary',
      },
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute('data-app', 'true');
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render dropdown button', async () => {
      mountWrapper();
      await wrapper.vm.$nextTick();

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render dropdown button menu', async () => {
      mountWrapper();

      wrapper.find(`#${TarkeebPlaces.fil}`).trigger('click');
      await wrapper.vm.$nextTick();

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('setFaail', () => {
    it('should emit event with null argument when faail is present', () => {
      mountWrapper();
      wrapper.vm.setFaail(HiddenPronounsDropdownText.faailPresent);

      expect(wrapper.emitted().faailSet![0][0]).toBeNull();
    });

    it('should emit event with argument when faail is selected', () => {
      mountWrapper();
      wrapper.vm.setFaail(HiddenPronounsDropdownText.huwa);

      expect(wrapper.emitted().faailSet![0][0]).toEqual(
        HiddenPronounsDropdownText.huwa
      );
    });
  });
});
