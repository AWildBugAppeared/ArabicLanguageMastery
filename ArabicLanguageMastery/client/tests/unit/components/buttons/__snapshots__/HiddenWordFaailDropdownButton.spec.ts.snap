// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenWordFaailDropdownButton.vue render should render dropdown button 1`] = `
"<div class=\\"v-menu\\"><button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary\\" id=\\"فعل\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"v-btn__content\\"><span class=\\"arabic\\">فعل</span></span></button>
  <!---->
</div>"
`;

exports[`HiddenWordFaailDropdownButton.vue render should render dropdown button menu 1`] = `
"<div class=\\"v-menu\\"><button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary\\" id=\\"فعل\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\"><span class=\\"v-btn__content\\"><span class=\\"arabic\\">فعل</span></span></button>
  <transition-stub name=\\"v-menu-transition\\"></transition-stub>
</div>"
`;
